mergify_engine/config.py

```diff
@@ -353,11 +376,49 @@ def ApplicationAPIKeys(v: str) -> typing.Dict[str, ApplicationAPIKey]:
 
 globals().update(Schema(CONFIG))
 
-if globals()["STREAM_URL"] is None:
-    STREAM_URL = globals()["STORAGE_URL"]
+# NOTE(sileht): legacy on premise installation before thing was auto sharded in redis databases
+# before on on premise STREAM/QUEUE/LEGACY_CACHE was in the same db, so keep until manuall migration
+# as been done
+if globals()["STORAGE_URL"] is not None:
+    for config_key in (
+        "DEFAULT_REDIS_URL",
+        "STREAM_URL",
+        "QUEUE_URL",
+        "LEGACY_CACHE_URL",
+    ):
+        if globals()[config_key] is None:
+            globals()[config_key] = globals()["STORAGE_URL"]
+
+
+# NOTE(sileht): If we reach 15, we should update onpremise installation guide
+# and add an upgrade release note section to ensure people configure their Redis
+# correctly
+REDIS_AUTO_DB_SHARDING_MAPPING = {
```

Comment:
I like the idea though OTOH this is not going to help if we ever decide to move a DB to a different Redis server 😢

Reply:
It just auto shard the default Redis, you can still move a particular database to another instance by settings explicitly QUEUE_URL or TEAM_MEMBERS_CACHE_URL

Reply:
I imagined another mechanism where we can map things like:

```
redis_servers:
  default: rediss://foo:1234
  user-defined-name: rediss://foo:1234
  queue: rediss://fish-and-chips:1234
  cache: rediss://bar-cache:1234

redis_db_mapping: // We provide defaults where server is always set to "default" and the default db is chosen by us.
  queue: 
     server: queue
     db: 3 
  legacy_cache: 
     server: cache
     db: 3
  team_members_cache: 
     server: cache
     db: 1
```

But I didn't find fancy way to have this in environment variables that works with Heroku.